bpf/lib/egress_gateway.h

```diff
@@ -260,4 +443,42 @@ int egress_gw_handle_packet(struct __ctx_buff *ctx,
 						NOT_VTEP_DST, trace);
 }
 
+static __always_inline
+int egress_gw_handle_packet_v6(struct __ctx_buff *ctx,
+                                struct ipv6_ct_tuple *tuple,
+                                __u32 src_sec_identity, __u32 dst_sec_identity,
+                                const struct trace_ctx *trace)
+{
+    struct endpoint_info *gateway_node_ep;
+    union v6addr gateway_ip = { 0 };
+    int ret;
+
+    /* If the packet is destined to an entity inside the cluster,
+     * either EP or node, it should not be forwarded to an egress
+     * gateway since only traffic leaving the cluster is supposed to
+     * be masqueraded with an egress IP.
+     */
+    if (identity_is_cluster(dst_sec_identity))
+        return CTX_ACT_OK;
+
+    ret = egress_gw_request_needs_redirect_hook_v6(tuple, &gateway_ip);
+    if (IS_ERR(ret))
+        return ret;
+
+    if (ret == CTX_ACT_OK)
+        return ret;
+
+    /* If the gateway node is the local node, then just let the
+     * packet go through, as it will be SNATed later on by
+     * handle_nat_fwd().
+     */
+    gateway_node_ep = __lookup_ip6_endpoint(&gateway_ip);
+    if (gateway_node_ep && (gateway_node_ep->flags & ENDPOINT_F_HOST))
+        return CTX_ACT_OK;
+
+    /* Send the packet to egress gateway node through a tunnel. */
+    return __encap_and_redirect_with_nodeid(ctx, &gateway_ip,
```

Comment:
@julianwiedmann This currently complains about the gateway_ip variable being of type union v6addr because __encap_and_redirect_with_nodeid expects a v4 address here. Do I need an additional call to lookup_ip6_remote_endpoint to retrieve the tunnel endpoint IP for the ipv6 gateway ip address?

Reply:
> Or does the gateway_ip continue to be an ipv4 address even with ipv6 policies?

yes! Supporting IPv6 node addresses would be a separate effort, right now we can focus on a dual-stack environment with IPv4 node-to-node connectivity. So for now all the gateway IPs would be IPv4.

Reply:
Thanks, I'll go fix that.